frontend/src/app/modules/boards/board/board-actions/version/version-action.service.ts

```diff
@@ -0,0 +1,223 @@
+      return (value instanceof HalResource) ? value.href! : value;
+    }
+
+    return;
+  }
+
+  /**
+   * Returns the loaded status
+   * @param query
+   */
+  public getLoadedFilterValue(query:QueryResource):Promise<undefined|VersionResource> {
+    const href = this.getFilterHref(query);
+
+    if (href) {
+      const id = HalResource.idFromLink(href);
+      return this.versionCache.require(id);
+    } else {
+      return Promise.resolve(undefined);
+    }
+  }
+
+  public addActionQueries(board:Board):Promise<Board> {
+    return this.getVersions()
+      .then((results) => {
+        return Promise.all<unknown>(
+          results.map((version:VersionResource) => {
+            if (version.isOpen() && version.definingProject.name === this.currentProject.name) {
+              return this.addActionQuery(board, version);
+            }
+
+            return Promise.resolve(board);
+          })
+        )
+          .then(() => board);
+      });
+  }
+
+  public addActionQuery(board:Board, value:HalResource):Promise<Board> {
+    let params:any = {
+      name: value.name,
+    };
+
+    let filter = {
+      version: {
+        operator: '=' as FilterOperator,
+        values: [value.id]
+      }
+    };
+
+    return this.boardListsService.addQuery(board, params, [filter]);
+  }
+
+  /**
+   * Return available versions for new lists, given the list of active
+   * queries in the board.
+   *
+   * @param board The board we're looking at
+   * @param queries The active set of queries
+   */
+  public getAvailableValues(board:Board, queries:QueryResource[]):Promise<HalResource[]> {
+    const active = new Set(
+      queries.map(query => this.getFilterHref(query))
+    );
+
+    return this.getVersions()
+      .then(results =>
+        results.filter(version => !active.has(version.href!))
+      );
+  }
+
+  /**
+   * Adds an entry to the list menu to edit the version if allowed
+   * @param {QueryResource} active query
+   * @returns {Promise<any>}
+   */
+  public getAdditionalListMenuItems(query:QueryResource):Promise<OpContextMenuItem[]> {
+    return this
+      .getLoadedFilterValue(query)
+      .then(version => {
+        if (version) {
+          return this.buildItemsForVersion(version);
+        } else {
+          return [];
+        }
+      });
+  }
+
+  public autocompleterComponent() {
+    return VersionAutocompleterComponent;
+  }
+
+  public headerComponent() {
+    return VersionBoardHeaderComponent;
+  }
+
+  public dragIntoAllowed(query:QueryResource, value:HalResource|undefined) {
+    return value instanceof VersionResource && value.isOpen();
+  }
+
+  private getVersions():Promise<VersionResource[]> {
+    if (this.currentProject.id === null) {
+      return Promise.resolve([]);
+    }
+
+    return this.versionDm
+      .listForProject(this.currentProject.id)
+      .then(collection => collection.elements);
+  }
+
+  private patchStatus(version:VersionResource, newStatus:'open'|'closed'|'locked') {
```

Comment:
Even at the risk of being too painstaking: The method name is a bit misleading considering the fact that we also have a status board. Maybe something like `patchVersionStatus` would be better.